pkg/contract_watcher/header/transformer/transformer.go

```diff
@@ -249,19 +253,32 @@ func (tr *Transformer) Execute() error {
 		}
```

Comment:
What if just moved all the stuff in this guard to the bottom of the loop that's iterating through missing headers? Seems like the below loops through `allLogs` and `sortedLogs` would be a no-op and we could delete some code

Reply:
A bit frazzled at the moment but I think I see what you mean and if I'm understanding correctly then the only thing that gives me pause is what to do if we run into an error for a header. We still want to mark the headers processed up until that point as checked which is why I was marking them as we go, but the much better approach would be to keep track of which index in the headers we are at when the error occurs and mark all those below it as checked in a single batch call. Looks like I've introduced quite a bit of unnecessary complexity here, thanks for catching that!

Reply:
Yeah I mean my thought was just that we'd call `MarkHeaderChecked` once per header, after we've iterated through all of its logs. [not proposing that we try to mark multiple headers checked at once, though I'm interested to check it out if it seems preferable]

I guess the likely objection would be making sure we preserve the ability to handle new contracts/logs being watched later on, but I think that would work fine? Assuming that a new checked column gets added and defaults to false...

Reply:
So after taking a closer look again, the issue there is that the header is normally checked marked in the same transaction that is used to persist logs in [this call](https://github.com/vulcanize/vulcanizedb/blob/staging/pkg/contract_watcher/header/transformer/transformer.go#L286). So what we have currently are calls to mark the header checked in the cases where that call is never reached.